### PR TITLE
feat: add Arrow IPC output format for http rest api

### PR DESCRIPTION
**Pull request by sunng87**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

This patch is initial work to add arrow output format for our http sql api. By using arrow output format it is possible to do zero-copy with arrowjs ecosystem, including visualization library vega (with vega-arrow-loader) and many more.

I will be seeking to add streaming support for output types like csv and arrow in future patch.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/servers/src/http/arrow_result.rs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    pub async fn from_output(mut outputs: Vec<crate::error::Result<Output>>) -> HttpResponse {
+        if outputs.len() > 1 {
+            return HttpResponse::Error(ErrorResponse::from_error_message(
+                ResponseFormat::Arrow,
+                StatusCode::InvalidArguments,
+                "Multi-statements are not allowed".to_string(),
+            ));
+        }
+
+        match outputs.remove(0) {
```
Comment (v0y4g3r) [inline src/servers/src/http/arrow_result.rs:72]: Line 64 only checks if  `outputs.len() > 1`, can `outputs` be empty here , which will cause panic?

Comment (sunng87) [reply src/servers/src/http/arrow_result.rs:72]: good catch

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**